src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -164,128 +144,102 @@ std::string parseSubscription(ConnectionInfo* ciP, ParseData* parseDataP, JsonDe
   }
   else if (!update)
   {
-    OrionError oe(SccBadRequest, "no notification for subscription specified");
-
-    alarmMgr.badInput(clientIp, "no notification specified");
-    return oe.render(ciP, "");
+    return error(ciP, "no notification for subscription specified");
   }
 
 
   // Expires field
-  if (document.HasMember("expires"))
-  {
-    const Value& expires = document["expires"];
+  Opt<std::string> expiresOpt = getStringOpt(document, "expires");
 
-    if (!expires.IsString())
-    {
-      OrionError oe(SccBadRequest, "expires is not a string");
-
-      alarmMgr.badInput(clientIp, "expires is not a string");
-      return oe.render(ciP, "");
-    }
+  if (!expiresOpt.ok())
```

Comment:
On the contrary, using "new fresh" checking methods would allow to have a more clear separation between NGSIv1 and NGSIv2 that could be benefitial in the long run.

To be discussed at skype.